anonymizer_module/opmon_anonymizer/opendata_anonymizer.py

```diff
@@ -0,0 +1,166 @@
+        self._allowed_fields = self._get_allowed_fields(field_translations_path, logger_manager)
+
+        hiding_rules = self._get_hiding_rules()
+        substitution_rules = self._get_substitution_rules()
+        transformers = self._get_transformers()
+
+        self._anonymization_job = (
+            OpenDataAnonymizationJob(
+                writer, hiding_rules, substitution_rules,
+                transformers, self._logger)
+            if not anonymization_job else anonymization_job
+        )
+
+    def anonymize(self, log_limit: Optional[int] = None) -> int:
+        writer_buffer_size = int(self._settings['postgres']['buffer-size'])
+        record_buffer = []
+
+        batch_start_mongodb_timestamp = None
+        last_successful_batch_timestamp = self._reader.last_processed_timestamp
+
+        processed_count = 0
+        for record in self._reader.get_records(self._allowed_fields):
+
+            if log_limit is not None and processed_count >= log_limit:
+                break
+
+            if batch_start_mongodb_timestamp is None:
+                batch_start_mongodb_timestamp = self._reader.last_processed_timestamp
+
+            record_buffer.append(record)
+
+            if len(record_buffer) >= writer_buffer_size:
+                batch_end_mongodb_timestamp = self._reader.last_processed_timestamp
+                batch_end_mongodb_timestamp = 1
+                try:
+                    self._anonymization_job.run(record_buffer)
+                except Exception:
+                    self._logger.log_error('failed_anonymizing_record_batch',
+                                           'Record batch with correctorTime within range [{0}, {1}] failed. '
+                                           'Last successful correctorTime was {2}'.format(
+                                               batch_start_mongodb_timestamp,
+                                               batch_end_mongodb_timestamp,
+                                               last_successful_batch_timestamp))
+                    self._reader.update_last_processed_timestamp(last_successful_batch_timestamp)
+                    return processed_count
+
+                processed_count += len(record_buffer)
+                record_buffer = []
+                self._logger.log_info(
+                    'record_batch_anonymized',
+                    f'{processed_count} records anonymized.'
+                    + f'correctorTime within range [{batch_start_mongodb_timestamp}, {batch_end_mongodb_timestamp}]'
+                )
+
+                batch_start_mongodb_timestamp = None
+                last_successful_batch_timestamp = batch_end_mongodb_timestamp
+
+        if record_buffer:
+            self._anonymization_job.run(record_buffer)
+            processed_count += len(record_buffer)
+
+        return processed_count
+
+    @staticmethod
+    def _get_allowed_fields(field_translations_file_path: str, logger: LoggerManager) -> list:
```

Comment:
Hmm, logger must be passed all the time? 🤔 looks strange… it should be global IMHO or at least given to instance on init time

Reply:
This is done due to `_get_allowed_fields()` being static method, which cannot access object attribute `self._logger`.
But it expected still to have all setup. I would leave it here for now.